Content.Server/_NF/PacifiedZone/PacifiedlZoneGeneratorSystem.cs

```diff
@@ -0,0 +1,94 @@
+using Robust.Server.GameObjects;
+using Robust.Shared.Timing;
+using Content.Shared.Humanoid;
+using Content.Shared.CombatMode.Pacification;
+using Content.Shared.Mind;
+using Content.Shared.Roles;
+using Content.Shared.Roles.Jobs;
+
+
+namespace Content.Server._NF.PacifiedZone
+{
+    public sealed class PacifiedZoneGeneratorSystem : EntitySystem
+    {
+        [Dependency] private readonly UserInterfaceSystem _userInterface = default!;
+        [Dependency] private readonly EntityLookupSystem _lookup = default!;
+        [Dependency] private readonly IGameTiming _gameTiming = default!;
+        [Dependency] private readonly IEntityManager _entMan = default!;
+        [Dependency] private readonly SharedMindSystem _mindSystem = default!;
+        [Dependency] private readonly SharedJobSystem _jobSystem = default!;
+
+        public override void Initialize()
+        {
+            base.Initialize();
+            SubscribeLocalEvent<PacifiedZoneGeneratorComponent, ComponentInit>(OnComponentInit);
+        }
+
```

Comment:
private void OnStartup(EntityUid uid, PacifiedByZoneComponentcomponent, ComponentStartup args)
    {
        if (!TryComp<CombatModeComponent>(uid, out var combatMode))
            return;

        if (component.DisallowDisarm && combatMode.CanDisarm != null)
            _combatSystem.SetCanDisarm(uid, false, combatMode);

        if (component.DisallowAllCombat)
        {
            _combatSystem.SetInCombatMode(uid, false, combatMode);
            _actionsSystem.SetEnabled(combatMode.CombatToggleActionEntity, false);
        }

        _alertsSystem.ShowAlert(uid, AlertType.Pacified);
    }

    private void OnShutdown(EntityUid uid, PacifiedByZoneComponentcomponent, ComponentShutdown args)
    {
        if (!TryComp<CombatModeComponent>(uid, out var combatMode))
            return;

        if (combatMode.CanDisarm != null)
            _combatSystem.SetCanDisarm(uid, true, combatMode);

        _actionsSystem.SetEnabled(combatMode.CombatToggleActionEntity, true);
        _alertsSystem.ClearAlert(uid, AlertType.Pacified);
    }
And accordingly, the logic for the new component